test/infrastructure/docker/api/v1beta2/devcluster_types.go

```diff
@@ -98,7 +97,11 @@ type DockerClusterBackendSpec struct {
 	// Instead, the docker cluster controller will simply copy these into the Status and allow the Cluster API
 	// controllers to do what they will with the defined failure domains.
 	// +optional
-	FailureDomains clusterv1beta1.FailureDomains `json:"failureDomains,omitempty"`
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
```

Comment:
I looked into the e2e tests and discussed this with Fabrizio. We have problem around conversion because the order of the array can change through conversion.

We would suggest the following:
* Add validation for all DevCluster / DockerCluster create/update calls that this array has to be alphabetically sorted by FailureDomain.Name
* Additionally we should adjust the conversion code so that when we convert to the array it is alphabetically sorted (similar to this: https://github.com/kubernetes-sigs/cluster-api/blob/495d454eb924254c9a8383a3747598183ae412cf/internal/contract/infrastructure_cluster.go#L161-L162)